cpp/src/sampling/detail/conversion_utilities.cu

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sampling/detail/sampling_utils.hpp"
+
+#include <cugraph/utilities/misc_utils.cuh>
+
+namespace cugraph {
+namespace detail {
+
+rmm::device_uvector<int32_t> convert_starting_vertex_offsets_to_labels(
+  raft::handle_t const& handle, raft::device_span<size_t const> starting_vertex_offsets)
+{
+  return expand_sparse_offsets(starting_vertex_offsets, int32_t{0}, handle.get_stream());
+}
+
+template <typename label_t>
+rmm::device_uvector<int32_t> flatten_label_map(
+  raft::handle_t const& handle,
+  std::tuple<raft::device_span<label_t const>, raft::device_span<int32_t const>>
+    label_to_output_comm_rank)
+{
+  rmm::device_uvector<int32_t> label_map(0, handle.get_stream());
+
+  label_t max_label = thrust::reduce(handle.get_thrust_policy(),
+                                     std::get<0>(label_to_output_comm_rank).begin(),
+                                     std::get<0>(label_to_output_comm_rank).end(),
+                                     label_t{0},
+                                     thrust::maximum<label_t>());
+
+  label_map.resize(max_label + 1, handle.get_stream());
```

Comment:
```
rmm::device_uvector<int32_t> label_map(0, handle.get_stream());
...
label_map.resize(max_label + 1, handle.get_stream());
```
=>
```
rmm::device_uvector<int32_t> label_map(max_label + 1, handle.get_stream());
```

Reply:
And shouldn't the caller already know # labels? Should we really compute this here?

Reply:
I believe we can infer the max/number of labels by looking at the size of the first device array in `label_to_output_comm_rank`. `max_label = std::get<0>(label_to_output_comm_rank) - 1` or 

`rmm::device_uvector<int32_t> label_map(label_to_output_comm_rank, handle.get_stream());`. @ChuckHastings is it a safe assumption to make?